src/C-interface/dense/bml_allocate_dense_typed.c

```diff
@@ -121,14 +146,29 @@ bml_matrix_dense_t *TYPED_FUNC(
     bml_matrix_dimension_t matrix_dimension = { N, N, N };
     bml_matrix_dense_t *A =
         TYPED_FUNC(bml_zero_matrix_dense) (matrix_dimension, distrib_mode);
+#ifdef BML_USE_MAGMA
+    MAGMA_T *A_dense = malloc(N * N * sizeof(REAL_T));
```

Comment:
Why another allocation? `A->matrix` should already be allocated in the previous line.

Reply:
First allocation is on GPU, second one on CPU. We initialize GPU data by copying data initialized on CPU. We should consider a GPU implementation of these functions if they are important and we need more efficiency.

Reply:
Do I understand this correctly then that you only need the CPU memory for initialization of the GPU data and from then on hold the matrix data in the GPU only?

Reply:
Correct.